src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -46,6 +47,17 @@ internal abstract class AbstractRazorDelegatingEndpoint<TRequest, TResponse> : I
     /// </summary>
     protected virtual bool PreferCSharpOverHtmlIfPossible { get; } = false;
 
+    /// <summary>
+    /// When <see langword="true"/>, we'll adjust the position of the caret to the name portion of an attribute if it is in the
+    /// attribute, but not the name. eg. if we get @bi$$nd-Value we'll act as though we got @bind-$$Value
+    /// </summary>
+    protected virtual bool TreatAnyAttributePositionAsAttributeName { get; } = false;
+
+    /// <summary>
+    /// If <see cref="TreatAnyAttributePositionAsAttributeName "/> is <see langword="true"/>, returns the range of the full attribute
+    /// </summary>
+    protected Range? OriginalAttributeRange { get; private set; }
```

Comment:
This seems very niche for this level of abstraction

Reply:
Yeah, definitely. This is where an "attribute based delegating handler" would be a good solution, but I get stuck against the fact that these handlers aren't _just_ attribute based, so it really is just one niche of their capabilities. Will have a think about it